Add Flipper wallet

cybercent opened the pull request:
The wallet has 2 versions:
- [Flipper](https://chrome.google.com/webstore/detail/flipper/lmmpaefggfcmnoddemmgdppddppnmhek) which runs on Mainnet
- [Flipper Developer](https://chrome.google.com/webstore/detail/flipper/nkjhfapihbllbhhneonipcinhdlcmfgn) which runs on Testnet 

_Note: Flipper Developer is pending Chrome store approval_

chasefleming commented:
Cool, thanks for adding this @cybercent  ! Will try it out. Which services does this support?

cybercent commented:
@chasefleming thanks, I looked through the docs and I have some questions, just be sure everything works as intended.

Our service is defined as below inside the extension:
```js
const service = {
    f_type: "Service",
    f_vsn: "1.0.0",
    type: "authn",
    uid: "flipper#authn",
    endpoint: "ext:0x4204aa9c92ab68a1",
    method: "EXT/RPC",
    id: "flipper",
    identity: {
        address: "0x4204aa9c92ab68a1",
    },
    provider: {
        address: "0x4204aa9c92ab68a1",
        name: "Flipper",
        icon: "data:image/svg+xml;base64,...",
        description: "...",
    },
}
```
but on the service discovery the definition is:
```js
{
       "f_type": "Service",
       "f_vsn": "1.0.0",
       "type": "authn",
       "method": "EXT/RPC",
       "uid": "flipper#authn",
       "endpoint": "chrome-extension://lmmpaefggfcmnoddemmgdppddppnmhek/index.html",
       "provider": {
         "address": "0x4204aa9c92ab68a1",
         "name": "Flipper",
         "icon": "data:image/svg+xml;base64,...",
         "description": "..",
         "color": "#6C52FF",
         "supportEmail": "...",
         "website": "..."
}
```

- Is the `service.id` needed?
- The docs mention that the `service.uid` should be unique, should the format always be`something` followed by `#authn`
- The `service.endpoint` is defined in the wallet example as `ext:` followed by your wallet's `address` but in the service discovery the chrome extension URL is used. Is that expected, or should they match ?

chasefleming commented:
Hi @cybercent excited about this PR. Re questions:

- `id` is not necessary, but won't affect anything being in
- `flipper#authn` as you have it is perfect
- `endpoint` should be what opens your app. Looks like the `ext:` is opening for me but the chrome extension link isn't

Also thanks for making a developer build in the extension store. Super helpful. One thing I noticed when testing is that after clicking "Confirm" on login multiple windows appear and nothing happens. Does that happen for you?

cybercent commented:
It looks like the multiple wallet popups are caused by FCL. 

We opened an issue that replicates using harness and demo wallet:
https://github.com/onflow/fcl-js/issues/1445

@chasefleming @gregsantos

gregsantos commented on data/services.json:
```diff
@@ -157,6 +174,23 @@
+      "type": "authn",
+      "method": "EXT/RPC",
+      "uid": "flipper-developer#authn",
+      "endpoint": "chrome-extension://nkjhfapihbllbhhneonipcinhdlcmfgn/index.html",
```
```suggestion
      "endpoint": "chrome-extension://nkjhfapihbllbhhneonipcinhdlcmfgn/index.html",
      "optIn": true,
```

gregsantos replied:
As Flipper does not currently support a user-signature service, you will need to specify "optIn": true.
dApps will be able to specify inclusion via fcl config if they do not require user-signature support.

gregsantos reviewed (changes requested):
It looks like you have resolved the pop up issues and are ready for inclusion in Discovery! 🚀
I tested mainnet and testnet extensions this morning and merged your Docs PR for available wallets. See review comments in regards making sure Flipper is an `opt-in` wallet until support for `user-signature` service is added.

chasefleming commented:
@cybercent PR looks great! Can you also add the metadata for the testnet address so the install link works on testnet?

chasefleming reviewed (changes requested):
@cybercent PR looks great! Can you also add the metadata for the testnet address so the install link works on testnet?

cybercent commented:
@chasefleming PTAL

- [x] I added the install link for the developer wallet
- [x] We released new wallet versions in the chrome store, that support message signing, so I removed the `optIn: true`

chasefleming commented:
@cybercent Love that you added user signing! Let's get this live. Great job.
